presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1532,6 +1536,32 @@ public void revokeTablePrivileges(ConnectorSession session, SchemaTableName sche
     }
 
     @Override
+    public List<GrantInfo> listTablePrivileges(ConnectorSession session, String schemaNameOrNull, String grantee)
+    {
+        ImmutableList.Builder<GrantInfo> grantMetadata = ImmutableList.builder();
+        for (String schemaName : listSchemas(session, schemaNameOrNull)) {
```

Comment:
You can use the metastore method `listTables(session, qualifiedTablePrefix)` for iterating over both schemas and tables. Also, how are views handled?